packages/sdk/src/cross-chain-provider.ts

```diff
@@ -436,4 +479,169 @@ export class CrossChainProvider implements ICrossChainProvider {
   ): Promise<number> {
     throw new Error('Not implemented')
   }
+
+  public async getChallengePeriodSeconds(): Promise<number> {
+    return BigNumber.from(
+      await this.contracts.l1.StateCommitmentChain.FRAUD_PROOF_WINDOW()
+    ).toNumber()
+  }
+
+  public async getChallengePeriodBlocks(): Promise<number> {
+    return Math.ceil(
+      (await this.getChallengePeriodSeconds()) / this.l1BlockTime
+    )
+  }
+
+  public async getMessageStateRoot(
```

Comment:
Everything from here down is basically copy/pasted from here: https://github.com/ethereum-optimism/optimism/blob/develop/packages/message-relayer/src/relay-tx.ts. I'll add tests for this at a later point in time.